internal/common/baseclient/iigo.go

```diff
@@ -56,14 +58,46 @@ func (c *BaseClient) GetClientSpeakerPointer() roles.Speaker {
 // COMPULSORY
 func (c *BaseClient) GetTaxContribution() shared.Resources {
 	// TODO: Implement common pool contribution greater than or equal to tax.
-	return 0
+	valToBeReturned := shared.Resources(0)
+	c.LocalVariableCache[rules.IslandTaxContribution] = rules.VariableValuePair{
+		VariableName: rules.IslandTaxContribution,
+		Values:       []float64{float64(valToBeReturned)},
+	}
+	isCompliant := c.CheckCompliance(rules.IslandTaxContribution)
+	if isCompliant {
+		// TODO: with this compliance check, agents can see whether they'd like to continue returning this value
+		valToBeReturned = 0
+	}
+	// Use the toolkit to recommend a value
+	newVal, success := c.GetRecommendation(rules.IslandTaxContribution)
+	if success {
+		// TODO: Choose whether to use this compliant value
+		valToBeReturned = shared.Resources(newVal.Values[0])
```

Comment:
What is nevVal.Values[0]? could the 0 be encoded as an enum maybe for clarity?

Reply:
Same on line 84 and 98

Reply:
Yeah good point, will do

Reply:
Done